src/LSST/control/PowerSubsystem.h

```diff
@@ -0,0 +1,187 @@
+          - U32 output voltage settling time (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 20ms
+          - U32 output voltage fall time (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 300ms
+          - DBL output voltage off level (double [64-bit real (~15 digit precision)] [V])  = 12V
+          - DBL maximum output current (double [64-bit real (~15 digit precision)] [A])  = 20A
+      - Comm Power Subsystem Configuration Information (typedef 'PowerSubsystem (cluster of 11 elements)
+            Values found in “PowerSubsystemCommonConfig.vi”  and “CommPowerBusConfigurationParameters.vi”
+         - U32 output on max delay (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])
+              “relay close delay” (50ms) + “breaker on time” (500ms) = 550ms
+         - U32 output off max delay (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])
+              “relay open delay” (30ms) = 30ms
+         - U32 reset breaker pulse width (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 400ms
+         - DBL breaker operating voltage (double [64-bit real (~15 digit precision)] [V])  = 19V
+         - CLUSTER output voltage warning level (volts) (cluster of 2 elements)  = “output voltage nominal level” = 24V
+           Based on “output voltage nominal level” * x%   x=”output voltage warning threshold level(%)”= 5%
+         - DBL Minimum (double [64-bit real (~15 digit precision)])  =  24V * 0.95 = 22.8
+         - DBL Maximum (double [64-bit real (~15 digit precision)])  = 24V * 1.05 = 25.2
+         -  CLUSTER output voltage fault level (volts) (cluster of 2 elements)
+            Based on “output voltage nominal level” * x%   x=”output voltage fault threshold level(%)”= 10%
+         - DBL Minimum (double [64-bit real (~15 digit precision)])  = 24V * 0.90 = 21.6
+         - DBL Maximum (double [64-bit real (~15 digit precision)])  = 24V * 1.10 = 26.4
+         - U32 breaker operating voltage rise time (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 30ms
+         - U32 output voltage settling time (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 10ms
+         - U32 output voltage fall time (ms) (unsigned long [32-bit integer (0 to 4,294,967,295)])  = 50ms
+         - DBL output voltage off level (double [64-bit real (~15 digit precision)] [V])  = 12V
+         - DBL maximum output current (double [64-bit real (~15 digit precision)] [A])  = 10A
+*/
+
+/// The motor power system and comm power system are the same except for configuration values. This class
+/// sets configuration values appropriately for each.
+/// Once set, these values are not expected to change.
+/// unit test: test_SimCore.cpp
+/// FUTURE: DM-40694 set the values using a configuration file.
+class PowerSubsystemConfig {
+public:
+    using Ptr = std::shared_ptr<PowerSubsystemConfig>;
+
+    enum SystemType {
+        MOTOR = 0,
+        COMM = 1
+    };
```

Comment:
You usually prefer the enum value in C++ to begin from 0 or 1? I was asked by my previous colleague before to begin from 1 all the time. What is your convention and we can stick to that in this `ts_m2cellcpp`. Thanks!

Reply:
Zero is the standard. I tend to hard code the indvidual values as C++ tends to log them as integers. It's a lot more reliable to pin down AXIAL = 27, than to count 27 items in.

In other cases, like the bitmaps, it's really important that they match the existing hardware, so if the existing hardware starts at 1, 1 will be the first enum, or 8 is that is the first value.